riposte-core/src/test/java/com/nike/riposte/server/componenttest/VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.java

```diff
@@ -0,0 +1,264 @@
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.nike.riposte.server.componenttest.VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.AfterSecurityRequestFilter.*;
+import static com.nike.riposte.server.componenttest.VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest.BeforeSecurityRequestFilter.*;
+import static java.util.Collections.singleton;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+
+@RunWith(DataProviderRunner.class)
+public class VerifyBeforeAndAfterSecurityRequestAndResponseFiltersComponentTest {
+
+    private static Server server;
+    private static ServerConfig serverConfig;
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        serverConfig = new RequestAndResponseFilterTestConfig();
+        server = new Server(serverConfig);
+        server.startup();
+    }
+
+    @AfterClass
+    public static void tearDown() throws Exception {
+        server.shutdown();
+    }
+
+    @Test
+    public void shouldExecuteBeforeAndAfterRequestAndResponseFilters() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpoint.MATCHING_PATH)
+                        .header("force-security-error", "false")
+                        .log().all()
+                        .when()
+                        .get()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode(), is(200));
+
+        assertThat(response.header(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
+        assertThat(response.header(AFTER_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(AFTER_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
+    }
+
+    @Test
+    public void shouldOnlyExecuteBeforeSecurityRequestFilterWhenSecurityErrorThrown() {
+        ExtractableResponse response =
+                given()
+                        .baseUri("http://127.0.0.1")
+                        .port(serverConfig.endpointsPort())
+                        .basePath(BasicEndpoint.MATCHING_PATH)
+                        .header("force-security-error", "true")
+                        .log().all()
+                        .when()
+                        .get()
+                        .then()
+                        .log().headers()
+                        .extract();
+
+        assertThat(response.statusCode(), is(401));
+
+        assertThat(response.header(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_KEY), is(BEFORE_SECURITY_FILTER_RESPONSE_HEADER_VALUE));
```

Comment:
The assertions are identical between these two tests, making it confusing. I wouldn't put assertions in the filters themselves - just have them set response headers so that the caller can determine what was called and what wasn't. That way the tests are clear without having to dig through the backing server code.